satip server: Rename nosatip to nosatipcli CLI flag

oliv3r opened the pull request:
The satip_boot function was added to config_boot in commit 442a871839bb ("satip server: fixed the RTP TCP size initialization, fixes #4517"), which doesn't tell us at all why it is there.

Whats worse, the nosatip flag seems to be ignored and the satip server still gets loaded.

Move things around a little so we can put them together under the nosatip flag to honor the intend better.

oliv3r commented:
Note, that I have not been able to properly test this, as I lack infrastructure to do so. Disabling satip with -nosatip seems to work now though according to debug prints.

Flole998 commented:
Let me fix the CI first, we still don't get all 45 checkmarks green.

Flole998 commented:
That flag is for the sat>ip client, you are now disabling the server. That seems not what is intended.

oliv3r commented:
> That flag is for the sat>ip client, you are now disabling the server. That seems not what is intended.

You sure? nasty double negations :)

The flag is 'nosatip' which would indicate to me, that it DISABLES the sat->ip server. So we check if nosatip is NOT set, and if it is not set, we start the server. Seems ok?

Flole998 commented:
That flag isn't related to the server at all. It is only for the client, and for that it works.

oliv3r commented:
Ohhh! server and CLIENT. Ok I'll rename the 'nosatip' flag to 'nosatipclient' for internal use, and see if it makes sense to add a flag for external.

The 'move' of the boot function should probably be in a better place though; not in the config section

oliv3r commented:
This is a brainfart really. I cleaned up the client stuff a little, and introduced the server flag.

If we feel the server flag is a bad idea, I can remove it, but keep the rest of this MR still.

Flole998 commented:
Isn't configuring the port as 0 already disabling the server?

Flole998 commented:
I think this is now also accessing an uninitialized variable....

oliv3r commented:
@Flole998 

I don't know, I am no satip user/expert. Can you elaborate on the uninitialized variable?

This is becoming more complicated then anticipated :D though the uninitialized variable sounds like a nastyness to begin with tbh

Flole998 commented:
I am not sure about the uninitialized variable, however you are overwriting the config with the default values apparently: In `satip_server_boot` there is
```
satip_server_conf.satip_deviceid = 1;
satip_server_conf.satip_rtptcpsize = 7896/188;
```
which would overwrite the settings previously loaded.

oliv3r commented:
Hmm, maybe that's why the boot function is in the middle of the config block? If we only had a commit message that explained what they where doing and _why_ :p

Flole998 commented:
> Hmm, maybe that's why the boot function is in the middle of the config block? If we only had a commit message that explained what they where doing and _why_ :p

To fix a bug, seems like the bug that you were about to reintroduce ;)

oliv3r commented:
lol, to work-around a bug :p ok fair nuff, not sure if I want to go through this effort right now :D
